momento-sdk/src/main/java/momento/sdk/CacheClient.java

```diff
@@ -677,6 +681,219 @@ public CompletableFuture<CacheSetFetchResponse> setFetch(String cacheName, Strin
+   *     at this rank will not be fetched. Defaults to null, which fetches up until and including
+   *     the last element.
+   * @param order - The order to fetch the elements in. Defaults to ascending.
+   * @return Future containing the result of the fetch operation.
+   */
+  public CompletableFuture<CacheSortedSetFetchResponse> sortedSetFetchByRank(
+      String cacheName,
+      String sortedSetName,
+      @Nullable Integer startRank,
+      @Nullable Integer endRank,
+      @Nullable SortOrder order) {
+    return scsDataClient.sortedSetFetchByRank(cacheName, sortedSetName, startRank, endRank, order);
+  }
+
+  /**
+   * Fetch all elements in the given sorted set by index (rank) in ascending order.
+   *
+   * @param cacheName - The cache containing the sorted set.
+   * @param sortedSetName - The sorted set to fetch from.
+   * @return Future containing the result of the fetch operation.
+   */
+  public CompletableFuture<CacheSortedSetFetchResponse> sortedSetFetchByRank(
+      String cacheName, String sortedSetName) {
+    return scsDataClient.sortedSetFetchByRank(cacheName, sortedSetName, null, null, null);
+  }
+
+  /**
+   * Fetch the elements in the given sorted set by score.
+   *
+   * @param cacheName - The cache containing the sorted set.
+   * @param sortedSetName - The sorted set to fetch from.
+   * @param minScore - The minimum score (inclusive) of the elements to fetch. Defaults to negative
+   *     infinity.
+   * @param maxScore - The maximum score (inclusive) of the elements to fetch. Defaults to positive
+   *     infinity.
+   * @param order - The order to fetch the elements in. Defaults to ascending.
+   * @param offset - The number of elements to skip before returning the first element. Defaults to
+   *     0. Note: this is not the rank of the first element to return, but the number of elements of
+   *     the result set to skip before returning the first element.
+   * @param count - The maximum number of elements to return. Defaults to all elements.
+   * @return Future containing the result of the fetch operation.
+   */
+  public CompletableFuture<CacheSortedSetFetchResponse> sortedSetFetchByScore(
+      String cacheName,
+      String sortedSetName,
+      @Nullable Double minScore,
+      @Nullable Double maxScore,
+      @Nullable SortOrder order,
+      @Nullable Integer offset,
+      @Nullable Integer count) {
+    return scsDataClient.sortedSetFetchByScore(
+        cacheName, sortedSetName, minScore, maxScore, order, offset, count);
+  }
+
+  /**
+   * Fetch all elements in the given sorted set by score in ascending order.
+   *
+   * @param cacheName - The cache containing the sorted set.
+   * @param sortedSetName - The sorted set to fetch from.
+   * @return Future containing the result of the fetch operation.
+   */
+  public CompletableFuture<CacheSortedSetFetchResponse> sortedSetFetchByScore(
```

Comment:
Should this take an offset and count since the other pared down fetch method already fetches the whole set?

Reply:
that's probably okay and we could add another signature without those in it in the future if we wanted, right?

Reply:
We could indeed.